[Tang Yetong] iP

darkv32 opened the pull request:
# Kewgy

### **Simple** chatbot made by me! Advantages:
- Simple
- Chatbot
- Easy to use (simple)

### Uses _file IO_ to make it better:

1. Store
2. Load
3. Save

### Uses OOP concepts, here is a snippet from the `main` method:

```java
public class Duke {

    public static void main(String[] args) {
        Storage storage = new Storage();
        Ui ui = new Ui();
        Parser parser = new Parser();
        List<Task> userTaskList = storage.loadTasks();
		...
	}
}
```

### TODO

- [X] CLI
- [ ] JavaFX

### An emoji

👻

### Block quote

> Here is a block quote

### Download

Download it from [here!](https://github.com/darkvoid32/ip/releases/tag/A-Jar)

###

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

darkv32 committed:
Level 0

darkv32 committed:
Level 1

darkv32 committed:
Level 2

darkv32 committed:
Level 3

darkv32 committed:
Level 4

darkv32 committed:
A-TextUiTesting

darkv32 committed:
Level-5

darkv32 committed:
Level 6

darkv32 committed:
A-Enums

darkv32 committed:
Level 7

darkv32 committed:
Merge tag 'Level-7'

darkv32 committed:
Level 8

darkv32 committed:
Merge branch 'branch-Level-8'

darkv32 committed:
A-MoreOOP

darkv32 committed:
Merge branch-A-MoreOOP

darkv32 committed:
A-Packages

darkv32 committed:
Merge branch-A-Packages

darkv32 committed:
Merge add-gradle-support

darkv32 committed:
A-JUnit

darkv32 committed:
Merge branch-A-JUnit

darkv32 committed:
A-JavaDoc

darkv32 committed:
A-CodingStandard

darkv32 committed:
Level-9

darkv32 committed:
Merge branch-A-JavaDoc

darkv32 committed:
Merge branch-A-CodingStandard

darkv32 committed:
Merge branch-Level-9

darkv32 committed:
Basic Echo with Kewqgy

darkv32 committed:
Added checkstyles

darkv32 committed:
Migration to JavaFX

darkv32 committed:
Printing now uses varargs to accept multi-line method input

darkv32 committed:
Merging branch A-CheckStyle

darkv32 committed:
Merging branch Level-10

darkv32 committed:
Merging branch A-Varargs

LimZiJia commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,53 @@
+                    return null;
+            }
+        }
+    }
```
Is this really necessary? Right now, all you use it for is to print the letters. I think it adds very little because the enum is not descriptive enough. For example instead of ```TaskType.T```, I think ```TaskType.ToDo``` would be more appropriate.

In the current iteration, I think even just replacing ```TaskType.X``` in the subclasses with the corresponding ```"X"``` might be better.

LimZiJia commented on src/main/java/Tasks/Deadline.java:
```diff
@@ -0,0 +1,50 @@
+
+    @Override
+    public String toString() {
+        return "[" + TaskType.D + "][" + super.getStatusIcon() + "] " + super.description;
```
Just a suggestion but you could explore using ```String.format()``` to make your code more readable. It is not that bad now because it is still short.
```suggestion
        return String.format("[%s][%s] %s\n", TaskType.D, super.getStatusIcon(), super.description);
```

LimZiJia commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,138 @@
+
+        String dukeText = "";
+
+        switch (nextCommand) {
```
Very nice use of enums and switch statements. This will be my inspiration to implement enums. Do note that the ```case``` and ```default``` statements should have the same indentation as the switch as per coding standard.

LimZiJia commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,138 @@
+                try {
+                    Deadline task = new Deadline(userMsgParsed[1]);
+                    userTaskList.add(task);
+                    storage.saveTask(userMsgParsed[1], TaskType.T);
```
There is a bug here!
```suggestion
                    storage.saveTask(userMsgParsed[1], TaskType.D);
```

darkv32 replied:
Good eye, thanks for finding it 😄

LimZiJia commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,138 @@
+                try {
+                    Event task = new Event(userMsgParsed[1]);
+                    userTaskList.add(task);
+                    storage.saveTask(userMsgParsed[1], TaskType.T);
```
Another bug here from ctrl + c 😆 
```suggestion
                    storage.saveTask(userMsgParsed[1], TaskType.E);
```

LimZiJia commented on src/main/java/Ui/Ui.java:
```diff
@@ -0,0 +1,202 @@
+            "Here are the matching tasks in your list:\n" + 
+            tasks + "\n"
+        );
+    }
```
I think with a more complicated function for ```userTaskList``` like this, it would be clearer to create a class ```TaskList```, so that in the future, we can easily add functions to the ```userTaskList``` instead of letting the UI complete this function.

LimZiJia reviewed (commented):
Overall great readability. There is one use of enums that I will take inspiration from and another that is not that useful (```TaskType```). Great work! 👍

darkv32 committed:
Added assertions into Parser class.

Variables passed into Parser class not checked if valid.

Assertions used to enforce various assumptions made about variables.

More assertions can be used in other classes as well in addition to
Parser.

darkv32 committed:
Current code base has slight inconsistencies in code quality

Small refactoring to improve code quality in various classes

Can continue with more intricate refactoring, currently only rough refactoring done

darkv32 committed:
Replaced some for loops with streams

Using Java Streams library to process some lists instead of a for loop.
This increases readability and other programmers can understand what
is going on in more concise lines

Other for loops can be processed the same way

darkv32 committed:
Merge pull request #1 from darkvoid32/branch-A-Assertions

Added assertions into Parser class.

darkv32 committed:
Merge branch 'master' into branch-A-CodeQuality

darkv32 committed:
Merge pull request #2 from darkvoid32/branch-A-CodeQuality

Improved code quality

darkv32 committed:
Merge branch 'master' into branch-A-Streams

darkv32 committed:
Merge pull request #3 from darkvoid32/branch-A-Streams

Replaced `for` loops with Java Streams

darkv32 committed:
Create gradle.yml

Added Github workflow `yml` file to run Github Actions

darkv32 committed:
Update gradle.yml

Updated `gradle.yml` to run tests and not code quality checks

darkv32 committed:
Update gradle.yml

Run only on `ubuntu`

darkv32 committed:
Merge pull request #4 from darkvoid32/darkvoid32-A-CI

Create `gradle.yml` to run Github Actions

darkv32 committed:
Added update time command

Allow users to update the time for deadlines and events using update_time command

Some bugs are not accounted for, edge cases and more robust parsing needed

darkv32 committed:
Fixed some bugs related to Event

Event was not printing dates properly after refactor

darkv32 committed:
Merge pull request #5 from darkvoid32/branch-C-Update

Added `update time` command

darkv32 committed:
Improved GUI

Added different color font for errors.
Increased window size for easier reading.
Added padding to labels to make it easier to read.

More can be done to improve GUI still

darkv32 committed:
Merge pull request #6 from darkvoid32/branch-A-BetterGui

Improved GUI

darkv32 committed:
Added more personality

Updated images to better suit chatbot and user
Give both different fonts, bolding and font size

More phrases can be given to the chatbot

darkv32 committed:
Merge pull request #7 from darkvoid32/branch-A-Personality

Added more personality

darkv32 committed:
Added more excaption handling

Error will be shown if file does not have read or write access when necessary:

darkv32 committed:
Merge pull request #8 from darkvoid32/branch-A-MoreErrorHandling

Added more excaption handling

darkv32 committed:
Refactored code

Changed from Duke to Kewgy to finalise chatbot name

darkv32 committed:
Added more tests

Added test for parser

Tests for storage can be added using mocking

darkv32 committed:
Merge pull request #9 from darkvoid32/branch-A-MoreTesting

Added more testing

darkv32 committed:
Added docs

darkv32 committed:
Added more java comments

darkv32 committed:
Updated main README

darkv32 committed:
Updated gitignore

darkv32 committed:
Small improvements

darkv32 committed:
Merge pull request #10 from darkvoid32/branch-A-Release

Small improvements

darkv32 committed:
Refactored code

Refactored based on comments by 2103 bot

darkv32 committed:
Merge branch 'master' of https://github.com/darkvoid32/ip
